### PR TITLE
Adapt to be used directly with fmu-ensemble

### PR DESCRIPTION
**Pull request by berland**
Based on what is needed to pass tests in fmu-ensemble

Necessary for able https://github.com/equinor/fmu-ensemble/issues/166

 * [x] Tests for the changes in ecl2df must be transferred from fmu-ensemble's test suite.
 * [x] Pylinting

Bonus changes:
* Deprecate non-datetime-index as return type from `df()`
* Refactor/deletion for code readability, normalize_dates() has been removed (the function is assumed to be private)

### CODE REVIEW

**Review by BryanLaura — APPROVED** [review]

File: ecl2df/eclfiles.py
Comment (jondequinor) [inline ecl2df/eclfiles.py:155]: Why is a failure a warning and not an actual failure?

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:8]: `dt` seems conventional, might be better here?

```python
import datetime as dt
…
```
But definitely unfortunate that `datetime` is a parameter 🤣

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:148]: Is swallowing the error here intentional? Maybe add a comment about why.

If not, maybe just add a `logger.debug` such that it is easier to debug.

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:258]: This is for presentational purposes I assume? Note that `[...]` is special notation in python, so I think
```python
f"{time_index_arg[0:3]} … {time_index_arg[-3:]}"
```

Might be less confusing?

File: ecl2df/eclfiles.py
Comment (berland) [reply ecl2df/eclfiles.py:155]: Right, degrading the failure should not happen here, but in fmu.ensemble if it wants to. Changing to raise.. from

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:8]: Changed to `dt`. Planning to remove the datetime parameter in the future.

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:148]: added comments.

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:258]: Suggestion incorporated!

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:73]: It doesn't hurt repeating this here.

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:75]: If parameters to `_drop_datelist` are the same as `resample_smry_dates` then these magic strings should really be shared between the two.

What exactly happens when `freq` is a date/datetime?

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:94]: Isn't it specifically the `freq` that is wrong here? Maybe raise 

```python
ValueError(f"freq argument must be this or that, not '{freq}'")
```

File: ecl2df/summary.py
Comment (jondequinor) [inline ecl2df/summary.py:113]: Document `first` as well?

File: ecl2df/summary.py
Comment (jondequinor) [reply ecl2df/summary.py:73]: Well, maybe it does. Ignore that comment.

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:75]: magic strings replaced by module constants. 

if `freq` is a datetime, it is caught in the `if`-statement ~6 lines below.

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:94]: Fixed.

File: ecl2df/summary.py
Comment (berland) [reply ecl2df/summary.py:113]: Fixed.

**Review by jondequinor — APPROVED** [review]
Technically this looks good to me. Can't really comment on the functional. Maybe you want a second pair of eyes?

**Comment by berland** [comment]
> Technically this looks good to me. Can't really comment on the functional. Maybe you want a second pair of eyes?

Thanks! I'll see if @asnyv  has any further comments before merge.

**Comment by asnyv** [comment]
I think it looks ok @berland 
Doesn't seem like we have tweaked on existing behavior (except future warning)? And existing tests ok.

**Pull request merged**
